docs/source/joblib.rst

```diff
@@ -52,6 +52,15 @@ validated parameter search as follows.
    with parallel_backend('dask.distributed', scheduler_host='localhost:8786'):
        search.fit(digits.data, digits.target)
 
+## Nested Parallelism
+
+Note that this code will not work if you distribute it via dask in order to implement nested parallelism. The reason is that each worker will have a fresh copy of sklearn and `from sklearn.externals.joblib import parallel_backend` will not apply to them, resulting in a "KeyError" when trying to distribute, but no problem when running the inner loop locally. The solution is 
+
+.. code-block:: python
+
+    from distributed.joblib import DistributedBackend 
+    register_parallel_backend('distributed', DistributedBackend, make_default=True)
```

Comment:
So the use-case is if you have an outer parallel loop which uses something like client.map(function, data) and then within your function there is a call to sklearn using dask.distributed:

```
import distributed.joblib

x = np.reshape(np.array(range(200)), (100,2))
y = np.random.normal(size=100)

def function(i):
    with parallel_backend('dask.distributed', scheduler_host=dask_addr):
        score = sk.model_selection.cross_val_score(sk.svm.SVR(kernel='linear') , x , y , cv=2, n_jobs=-1)
    return score 
```
here calling `function(1)` causes no problems. 

However if we now want to parallelize calls to `function(i)`:

    dask_client.gather(  dask_client.map(function, range(5))   )

we get `KeyError: 'dask.distributed'` in `sklearn/externals/joblib/parallel.py in parallel_backend()`. The fix is to put `parallel.register_parallel_backend('dask.distributed',DistributedBackend, make_default=True )` inside `function(i)`.